PalindromePartitioning.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * [WIP]
+ * Leetcode: No running
+ */
+
+class Solution {
+    List<List<String>> result;
+    public List<List<String>> partition(String s) {
+        result = new ArrayList<>();
+        //backtracking function
+        backtrack(s, 0, new ArrayList<>());
+        return result;
+    }
+    
+    private void backtrack(String s, int index, List<String> substrings){
+        if(index >= s.length() && substrings.size() > 0){
+            result.add(new ArrayList<>(substrings));
+            return;
+        } 
+         
+        for(int i = index; i < s.length(); i++){
+            String substring = s.substring(index, i);
```

Comment:
In substring, the end index is not included in the new string
so to make a string from index to i that range should be s.substring(index, i+1)